src/components/CriteriaProvider.tsx

```diff
@@ -61,13 +69,15 @@ function elmLibraryToCriteria(elm: ElmLibrary, custom = false): Criteria[] {
   });
 }
 
-function jsonToCriteria(rawElm: string): Criteria[] | undefined {
-  const elm = JSON.parse(rawElm);
-  if (!elm.library?.identifier) {
+function jsonToCriteria(rawCriteria: string): Criteria[] | undefined {
+  const criteria = JSON.parse(rawCriteria);
+  if (criteria.type) {
+    builderModelToCriteria(criteria);
+  } else if (!criteria.library?.identifier) {
```

Comment:
Did you intend to do something with this value? The result is not stored anywhere so the elmLibraryToCriteria function will be the result returned